src/operator/tensor/dot-inl.h

```diff
@@ -798,11 +776,16 @@ inline void DotCsrRspDnsImpl(const OpContext& ctx,
           if (trans_lhs) {
             LOG(FATAL) << "DotCsrRspDnsImpl has not implemented dot(csr.T, rsp) = dns yet";
           } else {
+            const RType* row_idx_ptr = row_idx_r.dptr<RType>();
+            std::unordered_map<RType, dim_t> row_idx_map;
+            for (dim_t ind = 0; ind < nnr; ind++) {
+              row_idx_map.emplace(row_idx_ptr[ind], ind);
```

Comment:
This may cause multiple rehashings at runtime. It's better to calculate the bucket count before defining the hashmap and use that number to initialize the hashmap. By default, it's the smallest prime number greater than the number of values.